### ISSUE TITLE
feat: physiology-driven AI survival behaviour (will to live)

### ISSUE THREAD

**Issue by MBarkerUK** [lErrorl404l/AEE#75]
## Real Thresholds Added (2026-09-16)

The survival-pressure model below is now calibrated against REAL military human-factors standards. Values come from TB MED 507, ISO 7243, NATO heat/cold stress guidance, Environment Canada/NWS wind chill, FAA hypoxia zones, and medical core-temperature references.

## Summary

AI in Arma 3 has a simple decision model: health, suppression, contact. It does not consider environment. This issue proposes a survival-prioritisation layer: AI makes decisions based on its physiological state - the will to live.

## Survival Pressure Calibration (REAL thresholds)

```
survivalPressure = max(dehydrationRisk, hypoxiaRisk, coldStress, heatStress, fatigue)
```

### Action bands (shared across systems)

| Pressure | Action |
|---|---|
| 0.0-0.3 | normal combat behaviour |
| 0.3-0.6 | seek cover, careful movement, calls for resupply |
| 0.6-0.8 | break contact, seek shelter/water/rest |
| 0.8-1.0 | abandon objective, route to survival |

### 1. Heat / WBGT (TB MED 507 flag bands, ISO 7243, NATO)

US Army flag categories and work/rest cycles (min, moderate/heavy/very heavy):

| Cat | WBGT C | Moderate | Heavy | Very heavy |
|---|---|---|---|---|
| 1 White | 25.6-27.7 | no limit | 40/20 | 20/40 |
| 2 Green | 27.8-29.4 | no limit | 30/30 | 15/45 |
| 3 Yellow | 29.4-31.1 | no limit | 30/30 | 10/50 |
| 4 Red | 31.1-32.2 | 50/10 | 20/40 | 10/50 |
| 5 Black | >32.2 | 20/40 | 15/45 | 10/50 |

Performance degradation (Foster 2021 PWC equation): WBGT 25 = 87% PWC, 30 = 67%, 33.6 = 50%, 35 = 44%, 40 = 25%. NATO work limit: core 38.5 C acclimatised, 38.0 C unacclimatised.

Mapping: <29 C = 0.0-0.3; 29-34 = 0.3-0.6; 34-38 = 0.6-0.8; >38 = 0.8-1.0

### 2. Wind chill (Environment Canada / NWS, TB MED 508)

| Band | C | Frostbite time |
|---|---|---|
| Low | 0 to -9 | none |
| Moderate | -10 to -27 | long exposure |
| High | -28 to -39 | 10-30 min |
| Very high | -40 to -47 | 5-10 min |
| Severe | -48 to -54 | 2-5 min |
| Extreme | <= -55 | <2 min |

Risk rises rapidly below -27 C. TB MED 508: hypothermia <35 C; memory registration impaired at 34.4-35.0 C.

Mapping: -28 = 0.3; -40 = 0.6; -48 = 0.8; -55 = 1.0

### 3. Dehydration (StatPearls, TB MED 507)

| Body water loss | Effect |
|---|---|
| 1-3% | mild, thirst |
| 2% | physical performance impaired |
| 3-5% | cognitive impaired |
| 4-6% | moderate, reduced work capacity |
| >6% | severe, collapse risk |

Fluid: 0.5-1.5 qt/hr (0.5-1.4 L/hr) by work load. Hard limits: max 1.5 qt/hr, max 12 qt/day.

Mapping: 2% = 0.3; 4% = 0.6; 6% = 0.8-1.0

### 4. Hypoxia (FAA AIM 8-1-1, CDC)

| Altitude | Effect |
|---|---|
| 2,500 m | cognitive/motor effects begin |
| 3,000-3,650 m | performance decrease (SpO2 70-65%) |
| 3,650-4,570 m | judgment/memory impaired; serious in 15 min at 4,570 m |
| 5,500 m | no acclimatisation possible |
| >7,500 m | death zone |

FAA: supplemental O2 above 10,000 ft day / 5,000 ft night; required above 14,000 ft.

Mapping: 3,000 m = 0.3; 3,650 = 0.45; 4,570 = 0.6; 5,500 = 0.8; 7,500 = 1.0

### 5. Fatigue (Thorne et al., DoDI 1010.10, ATP 6-22.5)

- 25% decline in useful mental work per 24 h awake
- 5-6 h sleep = blood alcohol 0.08%
- <5 h sleep = 20% decrease in thinking performance
- Standard: 7-8 h sleep per 24 h

NOTE: the common "1:3 work-rest ratio" is ACGIH heat research, not a military fatigue standard. Do not use it.

Mapping: 7 h = 0.0; 6 h = 0.25; 5 h = 0.5; <5 h = 0.7-1.0

### 6. Core temperature (StatPearls, TB MED 508, NATO)

| C | State | Pressure |
|---|---|---|
| <28 | severe hypothermia | 1.0 |
| 28-32 | moderate hypothermia | 0.7-0.9 |
| 32-35 | mild hypothermia | 0.3-0.6 |
| 35-36.5 | cold stress | 0.1-0.3 |
| 36.5-37.5 | normal | 0.0 |
| 37.5-38.5 | heat stress | 0.1-0.45 |
| 38.5-40 | heat illness | 0.45-0.75 |
| >40 | heat stroke | 0.75-1.0 |

### 7. JSP 912 (Human Factors Integration, V3.0 Mar 2024)

Management-process standard; no numeric thresholds. Mandates HFI via Def Stan 00-251; thermal hazards live in the System Safety domain. Real UK thermal numbers are in JSP 375 Vol 1 Ch 41 (Annex C work/rest tables, QuestTemp 34 monitor) - verify exact C bands against that PDF before shipping UK-specific values.

## Behaviour Model

```sqf
// Survival pressure 0..1, higher = more urgent
_survivalPressure = _max [
    aee_physiology_dehydrationRisk,   // 2% loss = 0.3, 6% = 1.0
    aee_physiology_hypoxiaRisk,       // 3000m = 0.3, 7500m = 1.0
    aee_physiology_coldStress,        // windchill -28 = 0.3, -55 = 1.0
    aee_physiology_heatStress,        // WBGT 29 = 0.3, >38 = 1.0
    aee_physiology_fatigueRisk        // 6h sleep = 0.25, <5h = 1.0
];
```

Environmental hazards add one-time urgency:
- Lightning risk > threshold -> AI avoids open ground
- Fire spread near -> AI evacuates downwind
- Flash flood risk -> AI moves to high ground

### Option A: Own addon (compat_ai)
- Reads aee_* variables, adjusts AI via existing AI mod surfaces
- Follows the six-compat pattern
- No core change

### Option B: Core integration
- AEE core gets an AI behaviour module
- Directly adjusts group behaviour, waypoints, commands
- Larger scope, deeper integration

**Recommendation**: Option A first (compat addon), Option B later if the behaviour model proves out.

## Validation

- Unit test the survival pressure function against EVERY band edge in the tables above (assert-based, no framework)
- Behaviour test: AI in cold+wind should seek shelter; AI in flood zone should route to high ground
- No per-frame reads - all at decision time

## Research References

- TB MED 507 (Heat Stress Control and Heat Casualty Management), US Army
- TB MED 508 (Cold Injury), US Army
- ISO 7243 (Hot environments - WBGT)
- Foster 2021 PWC equation: PWC% = 100/(1+(33.63/WBGT)^-6.33)
- Environment Canada / NWS wind chill index
- FAA AIM 8-1-1 (hypoxia altitude zones)
- NATO heat/cold stress guidance (ADA571324)
- JSP 912 V3.0 (Human Factors Integration, Mar 2024), Def Stan 00-251
- JSP 375 Vol 1 Ch 41 (UK thermal work/rest tables)
- DoDI 1010.10, ATP 6-22.5 (fatigue)
